### PR TITLE
RefreshLayout improvement

### PR DESCRIPTION
**Pull request by Kak0ytachel**
Added possibility to change color of RefreshSpinner and the circular ellipse around it via adding attributes `spinner_color` and `circle_color`

### CODE REVIEW

File: kivymd/uix/refreshlayout/refreshlayout.py [hunk 1]
```diff
@@ -150,7 +152,7 @@ def on_overscroll(
             return False
 
 
-class MDScrollViewRefreshLayout(MDScrollView):
+class MDScrollViewRefreshLayout(ThemableBehavior, MDScrollView):
```
Comment (HeaTTheatR) [inline kivymd/uix/refreshlayout/refreshlayout.py:155]: Why did you add the `ThemableBehavior` class as an heir?

File: kivymd/uix/refreshlayout/refreshlayout.py [hunk 2]
```diff
@@ -175,8 +177,26 @@ class MDScrollViewRefreshLayout(MDScrollView):
     and defaults to `None`.
     """
 
+    spinner_color = ColorProperty()
```
Comment (HeaTTheatR) [inline kivymd/uix/refreshlayout/refreshlayout.py]:
```python
spinner_color = ColorProperty(None)
```

File: kivymd/uix/refreshlayout/refreshlayout.py [hunk 3]
```diff
@@ -175,8 +177,26 @@ class MDScrollViewRefreshLayout(MDScrollView):
...
+    :attr:`spinner_color` is a :class:`~kivy.properties.ColorProperty`
+    and defaults to `[1, 1, 1, 1]`.
+    """
+
+    circle_color = ColorProperty(None)
+    """
+    Color of ellipse around the spinner in (r, g, b, a) or string format.
+
+    :attr:`circle_color` is a :class:`~kivy.properties.ColorProperty`
+    and defaults to `[1, 1, 1, 1]`.
```
Comment (HeaTTheatR) [inline kivymd/uix/refreshlayout/refreshlayout.py]:
```python
and defaults to `None`.
```

File: kivymd/uix/refreshlayout/refreshlayout.py [hunk 4]
```diff
@@ -187,7 +207,9 @@ def on_touch_up(self, *args):
             if self.refresh_callback:
                 self.refresh_callback()
             if not self.refresh_spinner:
-                self.refresh_spinner = RefreshSpinner(_refresh_layout=self)
+                self.refresh_spinner = RefreshSpinner(_refresh_layout=self,
```
Comment (HeaTTheatR) [inline kivymd/uix/refreshlayout/refreshlayout.py:210]: Use `pre-commit run --all-files`

File: kivymd/uix/refreshlayout/refreshlayout.py (hunk 1 above)
Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py:155]: In my opinion it's the most elegant solution possible here. Yeah, I could access the `ThemableBehavior.theme_cls.primary_dark` in `RefreshSpinner`, but in that case I had to  use some crutches like additional variable that can be `None` since `ColorProperty(None)` without reassignment causes crash

File: kivymd/uix/refreshlayout/refreshlayout.py (hunk 2 above)
Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: Do I have to reassign it if I'm absolutely satisfied with default value `[1, 1, 1, 1]` ?

File: kivymd/uix/refreshlayout/refreshlayout.py (hunk 1 above)
Comment (HeaTTheatR) [reply kivymd/uix/refreshlayout/refreshlayout.py:155]: You don't use `ThemableBehavior` class attributes anywhere

Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py:155]: I do use it in line 199 
``` py
        if not self.circle_color:
            self.circle_color = self.theme_cls.primary_dark
``` 
`theme_cls` is inherited from class `ThemableBehavior`

File: kivymd/uix/refreshlayout/refreshlayout.py (hunk 3 above)
Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: The same, there are no need to set `None` by default and later reassign `[1, 1, 1, 1]`

**Comment by HeaTTheatR** [comment]
@Kak0ytachel Pay attention to other comments...

**Comment by Kak0ytachel** [comment]
> @Kak0ytachel Pay attention to other comments...

Which of the comments exactly?

File: kivymd/uix/refreshlayout/refreshlayout.py [hunk 5]
```diff
@@ -210,6 +232,14 @@ class RefreshSpinner(ThemableBehavior, FloatLayout):
...
     """
 
+    circle_color = ColorProperty()
+    """
+    Color of ellipse around the spinner in (r, g, b, a) or string format.
+
+    :attr:`circle_color` is a :class:`~kivy.properties.ColorProperty`
+    and defaults to `[1, 1, 1, 1]`.
+    """
+
```
Comment (HeaTTheatR) [inline kivymd/uix/refreshlayout/refreshlayout.py]: Снимок экрана 2022-12-29 в 21 18 56

Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: yeah, my mistake, I have already fixed it in the second commit

File: kivymd/uix/refreshlayout/refreshlayout.py [hunk 6]
```diff
@@ -237,7 +237,7 @@ class RefreshSpinner(ThemableBehavior, FloatLayout):
     Color of ellipse around the spinner in (r, g, b, a) or string format.
 
     :attr:`circle_color` is a :class:`~kivy.properties.ColorProperty`
-    and defaults to `[1, 1, 1, 1]`.
+    and defaults to `theme_cls.primary_dark`.
```
Comment (HeaTTheatR) [inline kivymd/uix/refreshlayout/refreshlayout.py]: By default to None

Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: yeah, but by default if `MDScrollViewRefreshLayout.circlre_color` equals `None`, then it's reassigned to `theme_cls.primary_dark`, so `RefreshSpinner.circle_color` is also `theme_cls.primary_dark`

Comment (HeaTTheatR) [reply kivymd/uix/refreshlayout/refreshlayout.py]: Must specify the actual value.

Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: ok, here you are

Comment (HeaTTheatR) [reply kivymd/uix/refreshlayout/refreshlayout.py]:
```python
    spinner_color = ColorProperty()
    """
    Color of spinner in (r, g, b, a) or string format.

    :attr:`spinner_color` is a :class:`~kivy.properties.ColorProperty`
    and defaults to `[1, 1, 1, 1]`.
    """

    circle_color = ColorProperty(None)
    """
    Color of ellipse around the spinner in (r, g, b, a) or string format.

    :attr:`circle_color` is a :class:`~kivy.properties.ColorProperty`
    and defaults to `[1, 1, 1, 1]`.
    """
```

Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: you are probably viewing the first commit but should view the latest

Comment (HeaTTheatR) [reply kivymd/uix/refreshlayout/refreshlayout.py]: Please check your entire PR... You have a lot of typos there...

File: kivymd/uix/refreshlayout/refreshlayout.py (hunk 4 above)
Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py:210]: I don't get it, can you please explain?

File: kivymd/uix/refreshlayout/refreshlayout.py (hunk 6 above)
Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: uhhh.. typos like missed articles the or something else?

Comment (HeaTTheatR) [reply kivymd/uix/refreshlayout/refreshlayout.py]: https://user-images.githubusercontent.com/16930280/210007757-c0c7948e-ddab-47f1-be43-e9246af159a1.mov

Comment (Kak0ytachel) [reply kivymd/uix/refreshlayout/refreshlayout.py]: I have already told you about it. Default value of `ColorProperty()` without arguments is `[1, 1, 1, 1]`, so everything is fine with `spinner_color`. As for the `circle_color`, it's reassigned in line 199. Yeah, by default it's `None`, but cause it's for the docs, I strongly believe that it'll be better if `theme_cls.primary_darker` will remain

Comment (HeaTTheatR) [reply kivymd/uix/refreshlayout/refreshlayout.py]: Please correct the default values for the `spinner_color` and `circle_color` attributes
